### ISSUE TITLE
5.0 RC2 - Exception while indexing

### ISSUE THREAD

**Issue by niemyjski** [elastic/elasticsearch-net#2411]
Current reproduction steps: clone https://github.com/exceptionless/exceptionless/tree/feature/elastic5 and run the EventIndexTests.GetByBotAsync

Message:
```
"Token PropertyName in state Property would result in an invalid JSON object. Path ''."
```

Stack
```
   at Newtonsoft.Json.JsonWriter.AutoComplete(JsonToken tokenBeingWritten)
   at Newtonsoft.Json.JsonTextWriter.WritePropertyName(String name, Boolean escape)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeObject(JsonWriter writer, Object value, JsonObjectContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.Serialize(JsonWriter jsonWriter, Object value, Type objectType)
   at Nest.IndexRequestJsonConverter.WriteJson(JsonWriter writer, Object value, JsonSerializer serializer) in C:\Code\elasticsearch-net\src\Nest\Document\Single\Index\IndexRequestJsonConverter.cs:line 26
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeConvertable(JsonWriter writer, JsonConverter converter, Object value, JsonContract contract, JsonContainerContract collectionContract, JsonProperty containerProperty)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.Serialize(JsonWriter jsonWriter, Object value, Type objectType)
   at Newtonsoft.Json.JsonSerializer.SerializeInternal(JsonWriter jsonWriter, Object value, Type objectType)
   at Nest.JsonNetSerializer.Serialize(Object data, Stream writableStream, SerializationFormatting formatting) in C:\Code\elasticsearch-net\src\Nest\CommonAbstractions\SerializationBehavior\JsonNetSerializer.cs:line 81
   at Elasticsearch.Net.PostData`1.<WriteAsync>d__17.MoveNext() in C:\Code\elasticsearch-net\src\Elasticsearch.Net\Transport\PostData.cs:line 149
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at Elasticsearch.Net.HttpConnection.<RequestAsync>d__7`1.MoveNext() in C:\Code\elasticsearch-net\src\Elasticsearch.Net\Connection\HttpConnection.cs:line 174
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Elasticsearch.Net.RequestPipeline.<CallElasticsearchAsync>d__65`1.MoveNext() in C:\Code\elasticsearch-net\src\Elasticsearch.Net\Transport\Pipeline\RequestPipeline.cs:line 437
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Elasticsearch.Net.Transport`1.<RequestAsync>d__15`1.MoveNext() in C:\Code\elasticsearch-net\src\Elasticsearch.Net\Transport\Transport.cs:line 134
```

**Comment by niemyjski** [comment]
This was previously working, just seems odd that it would start failing, the json and object is valid.
<img width="1582" alt="screen shot 2016-11-23 at 3 40 11 pm" src="https://cloud.githubusercontent.com/assets/1020579/20579591/2af65da4-b193-11e6-9934-39ab8dc52a0f.png">

**Comment by niemyjski** [comment]
This seems to only happen when I have a property called reference id...

```json

 ```{
  "id": "54dbc16ca0f5c61398427b00",
  "organization_id": "1ecd0826e447ad1e78877555",
  "project_id": "1ecd0826e447ad1e78877ab2",
  "stack_id": "1ecd0826e447a44e78877ab1",
  "is_first_occurrence": false,
  "is_fixed": true,
  "is_hidden": false,
  "created_utc": "2015-02-11T20:54:04.4082284",
  "type": "log",
  "source": "GET Print",
  "date": "2015-02-11T20:54:04.3457274+00:00",
  "reference_id": "876554321"
}
```

I've gotten rid of all our serialization settings. I'll see if I can create a test.

**Comment by niemyjski** [comment]
I'm not sure if the following test will reproduce or not (can't run the tests :\ )

```cs
public class GithubIssue2411 : IClusterFixture<ReadOnlyCluster> {
        private readonly ReadOnlyCluster _cluster;

        public GithubIssue2411(ReadOnlyCluster cluster) {
            _cluster = cluster;
        }

        [I]
        public async Task NestedInnerHitsShouldIncludedNestedPropertyAsync() {
            var client = _cluster.Client;

            var response = await client.IndexAsync(new Events { ReferenceId = "12345678" }, i => {
                i.OpType(OpType.Create);
                i.Type("events");
                i.Index("events");
                return i;
            });

            response.ShouldBeValid();
        }

        public class Events {
            public string ReferenceId { get; set; }
        }
    }
```

**Comment by russcam** [comment]
The test runs but fails with the following:

```
Xunit.Sdk.XunitException
Expected True because Invalid NEST response built from a unsuccessful low level call on POST: /events/events?pretty=true&op_type=create
# Audit trail of this API call:
 - [1] BadResponse: Node: http://localhost:9200/ Took: 00:00:00.0700481
# ServerError: ServerError: 400Type: action_request_validation_exception Reason: "Validation Failed: 1: an id must be provided if version type or value are set;"
# OriginalException: System.Net.WebException: The remote server returned an error: (400) Bad Request.
   at System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization)
```

The request needs an id suppled because of the `op_type=create`

**Comment by niemyjski** [comment]
doh, yeah try adding a id property to the model and just set it to anything. I still can't run your tests.

**Comment by russcam** [comment]
I can reproduce the issue; it's related to the change in PR https://github.com/elastic/elasticsearch-net/pull/2399

**Comment by niemyjski** [comment]
Thanks for getting to the bottom of this... now I'll be able to sleep tonight. Will there be an rc3 real soon :)

**Comment by russcam** [comment]
This is fixed and in `5.0.0-rc3`
